src/test/java/org/sagebionetworks/web/unitclient/widget/discussion/SingleDiscussionThreadWidgetTest.java

```diff
@@ -768,6 +767,56 @@ public void testCheckForInViewAndLoadDataAttachedInViewAndVisible() {
 				any(DiscussionFilter.class), any(AsyncCallback.class));
 	}
 
+	@Test
+	public void testOnClickCancel() {
+		discussionThreadWidget.onClickCancel();
+		verify(mockView).resetButton();
+		verify(mockView).setReplyTextBoxVisible(true);
+		verify(mockView).setNewReplyContainerVisible(false);
+	}
+
+	@Test
+	public void testOnClickSaveInvalidArgument() {
+		when(mockMarkdownEditorWidget.getMarkdown()).thenReturn("");
+		discussionThreadWidget.onClickSave();
+		verify(mockSynAlert).clear();
+		verify(mockMarkdownEditorWidget).getMarkdown();
+		verify(mockSynAlert).showError(anyString());
+		verifyZeroInteractions(mockDiscussionForumClientAsync);
+	}
+
+	@Test
+	public void testOnClickSaveSuccess() {
+		when(mockMarkdownEditorWidget.getMarkdown()).thenReturn("message");
+		AsyncMockStubber.callSuccessWith(mockDiscussionReplyBundle)
+			.when(mockDiscussionForumClientAsync).createReply(any(CreateDiscussionReply.class),
+					any(AsyncCallback.class));
+		discussionThreadWidget.onClickSave();
+		verify(mockSynAlert, atLeastOnce()).clear();
+		verify(mockView).showSaving();
+		verify(mockView).showSuccess(anyString(), anyString());
+		verify(mockDiscussionForumClientAsync).createReply(any(CreateDiscussionReply.class), any(AsyncCallback.class));
+		verify(mockView).resetButton();
+		verify(mockView).setReplyTextBoxVisible(true);
+		verify(mockView).setNewReplyContainerVisible(false);
+		verify(mockDiscussionForumClientAsync).getThread(anyString(), any(AsyncCallback.class));
+	}
+
+	@Test
+	public void testOnSaveFailure() {
+		when(mockMarkdownEditorWidget.getMarkdown()).thenReturn("message");
+		AsyncMockStubber.callFailureWith(new Exception())
+			.when(mockDiscussionForumClientAsync).createReply(any(CreateDiscussionReply.class),
+					any(AsyncCallback.class));
+		discussionThreadWidget.onClickSave();
+		verify(mockSynAlert).clear();
+		verify(mockView).showSaving();
+		verify(mockDiscussionForumClientAsync).createReply(any(CreateDiscussionReply.class), any(AsyncCallback.class));
+		verifyZeroInteractions(mockCallback);
+		verify(mockSynAlert).handleException(any(Throwable.class));
```

Comment:
You could verify that the syn alert is given the exception that comes back from the rpc.